src/app/admin/components/filter-search/filter-search.component.ts

```diff
@@ -0,0 +1,267 @@
+   */
+  toggleCollectionMenu(event: MouseEvent) {
+    if (this.collectionMenu) {
+      if (!this.collectionMenuDown && event) {
+        this.contextMenuService.open(
+          this.collectionMenu,
+          event.currentTarget as HTMLElement,
+          {
+            top: 10,
+            left:
+              (event.currentTarget as HTMLElement).getBoundingClientRect()
+                .width - 200
+          }
+        );
+      } else {
+        this.contextMenuService.destroy(this.collectionMenu);
+      }
+
+      this.collectionMenuDown = !!event;
+    } else {
+      console.error('Error! Attempted to use an unregistered context menu');
+    }
+  }
+
+  /**
+   * Add or remove filter from filters list
+   * @param filter {string} the filter to be toggled
+   */
+  toggleStatusFilter(filter: string) {
+    if (filter.toLowerCase() === 'all') {
+      this.clearStatusFilters();
+      this.toggleFilterMenu(undefined);
+      return;
+    }
+
+    if (this.filters.has(filter)) {
+      this.filters.delete(filter);
+    } else {
+      this.filters.add(filter);
+    }
+    this.statusFilter.emit(Array.from(this.filters));
+  }
+
+  /**
+   * Add or remove filter from filters list
+   * @param filter {string} the filter to be toggled
+   */
+  toggleCollectionFilter(filter: string) {
+    if (filter.toLowerCase() === 'all') {
+      this.clearCollectionFilters();
+      this.toggleCollectionMenu(undefined);
+      return;
+    } else if (
+      this.selectedCollection &&
+      filter === this.selectedCollection.abvName
+    ) {
+      this.clearCollectionFilters();
+    } else {
+      this.setSelectedCollection(filter);
+      this.collectionFilter.emit(filter);
+    }
+  }
+
+  /**
+   * Remove all applied status filters
+   */
+  clearStatusFilters() {
+    this.filters.clear();
+    this.statusFilter.emit([]);
+  }
+
+  /**
+   * Remove any applied collection filters
+   *
+   * @memberof FilterSearchComponent
+   */
+  clearCollectionFilters() {
+    this.setSelectedCollection(undefined);
+    this.collectionFilter.emit(undefined);
+  }
+
+  /**
+   * Remove all active status filters and collection filters
+   *
+   * @memberof FilterSearchComponent
+   */
+  clearAllFilters() {
+    this.setSelectedCollection(undefined);
+    this.filters.clear();
+    this.clearAll.emit();
+  }
+
+  /**
+   * Returns the correct icon for the given Learning Object status
+   *
+   * @param {string} status the status for which to return an icon
+   * @returns {string}
+   * @memberof FilterSearchComponent
+   */
+  getStatusIcon(status: string): string {
+    switch (status) {
+      case 'unreleased':
+        return 'far fa-eye-slash';
+      case 'waiting':
+        return 'far fa-hourglass';
+      case 'review':
+        return 'far fa-sync';
+      case 'proofing':
+        return 'far fa-shield';
+      case 'released':
+        return 'far fa-eye';
+      case 'rejected':
+        return 'far fa-ban';
+    }
```

Comment:
Could we pull this up to a higher level of the application where it could be shared across components/modules? It seems to be popping up a lot, so any changes will require updates in several places.

Reply:
As far as I'm aware, this is the only place we're doing the icons like this. Perhaps some time should be taken to decide the best way to do the status icons across the platform, but as of right now I can't find any code that performs a similar function to this in the app.

Reply:
I was thinking of `LearningObjectStatusIndicatorComponent` in my head, but it turns out I never went back to refactor the template so it's just making a ton of icon tags and ng-ifing them